### ISSUE TITLE
IPythonLexer test failure with pygments 2.1

### ISSUE THREAD

**Issue by takluyver** [ipython/ipython#9201]
Jenkins is showing us a test failure on Windows:

https://jenkins.jupyter.org/job/windows-multi/_pyversion=3.5,label=windows/504/testReport/

It looks like `$HOME` is unexpectedly being split into two tokens. I suspect it's failing since pygments 2.1 was released. I see the same locally on Linux, but it's not failing on Travis (maybe we're not installing pygments so those tests are skipped?).

@abalkin and @chebee7i have done the most signficant work on the lexer machinery. Any ideas?

**Comment by minrk** [comment]
This is a change in the BashLexer in pygments 2.1

Running this:

```python
import sys
import pygments
from pygments.lexers import BashLexer

print('Python-%s' % sys.version.split()[0])
print('Pygments-%s' % pygments.__version__)

lexer = BashLexer()
print(list(lexer.get_tokens('$HOME')))
```

Gives:

```
Python-3.5.1
Pygments-2.1
[(Token.Text, '$'), (Token.Text, 'HOME'), (Token.Text, '\n')]
```

```
Python-3.5.1
Pygments-2.0.2
[(Token.Name.Variable, '$HOME'), (Token.Text, '\n')]
```

Same results on Python 2.

Seems like a bug in pygments.

**Comment by minrk** [comment]
I think this is fixed by #9202, which checks against BashLexer's actual output, rather than assuming we know what BashLexer's output will be.
